src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java

```diff
@@ -1,44 +1,278 @@
 public class SubscriptionTriggerSteps extends FunctionalTestBase {
 
+    private static final String SUBSCRIPTION_WITH_JSON_PATH = "src/functionaltests/resources/subscription_multiple.json";
+    private static final String EIFFEL_EVENTS_JSON_PATH = "src/functionaltests/resources/eiffel_events_for_test.json";
+
+    private static final String REST_ENDPOINT = "/rest";
+    private static final String REST_ENDPOINT_AUTH = "/rest/with/auth";
+    private static final String REST_ENDPOINT_PARAMS = "/rest/with/params";
+    private static final String REST_ENDPOINT_AUTH_PARAMS = "/rest/with/auth/params";
+    private static final String BASE_URL = "localhost";
+
+    private static ObjectMapper objectMapper = new ObjectMapper();
+
+    @Value("${email.sender}")
+    private String sender;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private RmqHandler rmqHandler;
+
+    @Autowired
+    private JavaMailSenderImpl mailSender;
+
+    private MvcResult result;
+    private SimpleSmtpServer smtpServer;
+    private ClientAndServer restServer;
+    private MockServerClient mockClient;
+
     private static final Logger LOGGER = LoggerFactory.getLogger(SubscriptionTriggerSteps.class);
 
-    @Given("^Subscription rest trigger is created$")
-    public void subscription_rest_trigger_is_created() throws Throwable {
-        LOGGER.debug("Creating subscription rest trigger");
-        System.out.println("MongoDB port for " + this.getClass().getName() + " is: " + getMongoDbPort());
+    @Before("@SubscriptionTriggerScenario")
+    public void beforeScenario() {
+        LOGGER.debug("Starting SMTP and REST Mock Servers");
+        try {
+            int port = SocketUtils.findAvailableTcpPort();
+            LOGGER.debug("Setting SMTP port to " + port);
+            mailSender.setPort(port);
+            smtpServer = SimpleSmtpServer.start(port);
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+
+        int port = SocketUtils.findAvailableTcpPort();
+        LOGGER.debug("Setting REST port to " + port);
+        restServer = startClientAndServer(port);
+        
+        // Set up endpoints
+        mockClient = new MockServerClient(BASE_URL, port);
+        mockClient.when(request().withMethod("POST").withPath(REST_ENDPOINT))
+                .respond(response().withStatusCode(201));
+        mockClient.when(request().withMethod("POST").withPath(REST_ENDPOINT_AUTH))
+                .respond(response().withStatusCode(201));
+        mockClient.when(request().withMethod("POST").withPath(REST_ENDPOINT_PARAMS))
+                .respond(response().withStatusCode(201));
+        mockClient.when(request().withMethod("POST").withPath(REST_ENDPOINT_AUTH_PARAMS))
+                .respond(response().withStatusCode(201));
+    }
+
+    @After("@SubscriptionTriggerScenario")
+    public void afterScenario() {
+        LOGGER.debug("Stopping SMTP and REST Mock Servers");
+        smtpServer.stop();
```

Comment:
add both smtpServer and restServer stop inside try.

Reply:
done